magento/pwa-studio
# Lazy Load Category Filters

## Description

The category page's load performance looks like:

![Screen Shot 2019-11-13 at 9 18 58 AM](https://user-images.githubusercontent.com/13182778/68787309-de74fa00-0606-11ea-9be5-4a78c13c5a3a.png)

As you can see, over half the time spent rendering `Category` is spent rendering the `FilterModal`. There weren't any major optimizations to make therein (it just takes time), so instead this PR lazy loads the `FilterModal` ~~when the user hovers, focuses, or clicks the "Filter" button~~:

![Screen Shot 2019-11-13 at 11 19 59 AM](https://user-images.githubusercontent.com/13182778/68787659-8db1d100-0607-11ea-83f5-c612fdcbaee0.png)

The new performance profile looks like this:

![Screen Shot 2019-11-13 at 9 43 29 AM](https://user-images.githubusercontent.com/13182778/68787473-3b70b000-0607-11ea-992b-e7b0068ba042.png)

Notice that `Category` is finished rendering much quicker now, and `FilterModal` is lazy loaded in after the fact.

~~Bonus: if the user never interacts with the "Filters" button, we never download the `FilterModal` chunk 🎉~~
Note: the above is still supported, but Venia defaults to loading the chunk immediately instead of waiting for hover / focus / click interactions. Developers who want this functionality can easily add it with a one-line change (it is documented how to do so in the code itself).

## Related Issue





PWA-55.

## Acceptance 



### Verification Stakeholders


@jimbo @sirugh @tjwiebell @revanth0212 

### Specification



### Verification Steps


1. Go to a category page in Venia
2. Open your browser Network tab
3. Hover over, focus, or click the "Filter" button
4. See that then and only then are the chunks for `FilterModal` downloaded (you can inspect their contents to see what components are loaded).

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
